### DISCUSSION TITLE
sealed trait MatchResult

### DISCUSSION THREAD

**Discussion by lars-n** [etorreborre/specs2#1047] [General]
Hi,

I'm currently in the process of upgrading a code base which has some non-test code extending and using `MatchResult` directly. Just wondering if there was any specific reason why the trait `MatchResult` ended up being sealed (besides preventing what I encountered 😏  )?

**Comment by etorreborre** [comment]
The reason is that they are various case classes implementing that trait and we need to pattern match exhaustively on them. Note that this trait actually disappears in the next major version of specs2 to simply use `Result`

**Comment by lars-n** [comment]
Thanks for the reply. Guess there's some refactoring coming up here
